tests/ErrorTest.php

```diff
@@ -199,8 +198,15 @@ public function testTwigArrayMapThrowsRuntimeExceptions()
         $out = $template->render(['variable' => [1, 2, 3, 4]]);
         $this->assertEquals('36912', trim($out));
 
+        set_error_handler(function ($errno, $errstr, $errfile, $errline): never {
+            throw new \ErrorException($errstr, $errno, $errfile, $errline);
+        });
```

Comment:
It is my understanding that this test only ever worked because PHPUnit 9 registered an error handler that throws on PHP warnings. Not sure if that was actually intended.

Reply:
To me, the right fix is to fix the implementation of `map` to check the type of variable as done in `reduce`: https://github.com/twigphp/Twig/blob/3.x/src/Extension/CoreExtension.php#L1834

Reply:
We could do that. Shall I submit that as a bugfix for 3.x?

Reply:
see #4272